docker/init/scripts/init.py

```diff
@@ -0,0 +1,417 @@
+		else:
+			for path in ttl_paths:
+				upload_file_to_s3(session, path, bucket, bucket_prefix)
+
+			# create the batch job information
+			job['S3_SUBMISSION_BUCKET'] = bucket
+			job['S3_SUBMISSION_PREFIX'] = bucket_prefix
+			job['VALIDATION_FLAGS'] = validation_type['validation']
+			job['S3_SUBMISSION_VALIDATION_DESCR'] = validation_type['description']
+			job['S3_SUBMISSION_EXTRACTED'] = len(ttls)
+
+	return job
+
+def check_for_duplicates(ttls):
+	"""Function will check for duplicates in a list of file names.
+
+	:param List ttls: A list of ttl file names.
+	:returns: True if duplicates were found, False otherwise
+	:rtype: bool
+	"""
+	if len(ttls) != len(set(ttls)):
+		logging.error("Duplicate files with .ttl extension found in submission")
+		return True
+	return False
+		
+
+def check_nist_directory(directory):
+	"""Helper function that will determine if NIST directory exists as an 
+	immediate subdirectory of the passed in directory.
+
+	:param str directory: The directory to validate against
+	:returns: True if directory exists, False otherwise
+	:rtype: bool
+	"""
+	return os.path.exists(directory + "/" + NIST['directory'])
+
+
+def check_inter_ta_directory(directory):
+	"""Helper function that will determine if INTER-TA directory exists as
+	an immediate subdirectory of the passed in directory.
+
+	:param str directory: The directory to validate against
+	:returns: True if directory exists, False otherwise
+	:rtype: bool
+	"""
+	return os.path.exists(directory + "/" + INTER_TA['directory'])
+
+
+def is_env_set(env, value):
+    """Helper function to check if a specific environment variable is not None
+
+    :param str env: The name of the environment variable
+    :param value: The value of the environment variable
+    :returns: True if environment variable is set, False otherwise
+    :rtype: bool
+    """
+    if not value:
+        logging.error("Environment variable %s is not set", env)
+        return False
+
+    logging.info("Environment variable %s is set to %s", env, value)
+    return True
+
+
+def validate_envs(envs):
+    """Helper function to validate all of the environment variables exist and are valid before
+    processing starts.
+
+    :param dict envs: Dictionary of all environment variables
+    :returns: True if all environment variables are valid, False otherwise
+    :rtype: bool
+    """
+    if not bool(envs):
+        logging.error("No environment variables found")
+        return False
+
+    for k, v in envs.items():
+        if not is_env_set(k, v):
+            return False
+
+    return True
+
+
+def read_envs():
+	"""Function will read in all environment variables into a dictionary
+
+	:returns: Dictionary containing all environment variables or defaults
+	:rtype: dict
+	"""
+	envs = {}
+	envs['S3_SUBMISSION_ARCHIVE_PATH'] = os.environ.get('S3_SUBMISSION_ARCHIVE', 'aida-validation/archives/NextCentury_1.zip')
```

Comment:
S3_SUBMISSION_ARCHIVE_PATH always sets to the default.  S3_SUBMISSION_ARCHIVE should be changed to avoid this bug.  Also, you might want to remove the default values in this and the main/worker script.